cryptography/src/bls12381/scheme.rs

```diff
@@ -49,6 +51,9 @@ impl Scheme for Bls12381 {
             Ok(key) => key,
             Err(_) => return None,
         };
+        if private_key == ZERO_PRIVATEKEY_BYTES {
+            return None;
+        }
```

Comment:
vector 10 says:
```
private_key: "0x0000000000000000000000000000000000000000000000000000000000000000"
message: "0xabababababababababababababababababababababababababababababababab"
signature: ""
```

Unlike the others, this given test vector is not a generated output from the python code, it is an hardcoded json.
With the project implementation, a non empty signature is returned:
`\xc0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0`

The change makes the BLS12381 factory failing when providing an empty private key. With this rule, we end with the `signature: ""` as we cannot sign.

Reply:
I'm actually surprised that `blst_scalar_fr_check` isn't sufficient here. 🤔

Reply:
I did a quick test and verified replacing `blst_scalar_fr_check` with `blst_sk_check` returns `None` properly.

Reply:
Looks like this was discussed here (on page 12): https://static1.squarespace.com/static/5eb0c72a0914c75f344e55db/t/5fd3f2a0c22572489c5ca171/1607725730065/NCC+Group+Audit+-+blst.pdf

`blst_scalar_fr_check` does not enforce the zero check and `blst_sk_check` was added for this case.

Reply:
Here is the corresponding commit (where they updated their bindings): https://github.com/supranational/blst/commit/15f63834dad3a3d9588574b302669f34d9f252bc

Let's do the same and just replace `blst_scalar_fr_check` with `blst_sk_check`.

Reply:
For those curious, here are the underlying implementations: https://github.com/supranational/blst/blob/bef14ca512ea575aff6f661fdad794263938795d/src/exports.c#L107-L113